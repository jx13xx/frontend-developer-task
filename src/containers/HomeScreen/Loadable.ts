/**
 *  Asynchronously load the Home Screen Component
 */
import loadable from "../../utils/loadable";
export default loadable(() => import("./index"));
