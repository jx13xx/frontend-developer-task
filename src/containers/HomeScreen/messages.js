import { defineMessages } from "react-intl";

export default defineMessages({
  headingText: {
    id: "homescreen.heading.text",
    defaultMessage: "Become a Member",
  },
  descriptionText: {
    id: "homescreen.description.text",
    defaultMessage: "Enter your valid info to get registered",
  },
  fullNameLabel: {
    id: "homescreen.fullName.label",
    defaultMessage: "Full Name",
  },
  fullNamePlaceHolder: {
    id: "homescreen.fullName.placeHolder",
    defaultMessage: "Your full name here",
  },
  emailAddressLabel: {
    id: "homescreen.emailAddress.label",
    defaultMessage: "Email Address",
  },
  emailAddressPlaceHolder: {
    id: "homescreen.emailAddress.placeHolder",
    defaultMessage: "Your email address here",
  },
  companyLabel: {
    id: "homescreen.company.label",
    defaultMessage: "Company",
  },
  companyLabelPlaceHolder: {
    id: "homescreen.company.placeHolder",
    defaultMessage: "Your Company",
  },
  companyWebsiteLabel: {
    id: "homescreen.companyWebsite.label",
    defaultMessage: "Company Website",
  },
  companyWebsitePlaceHolder: {
    id: "homescreen.companyWebsite.placeHolder",
    defaultMessage: "Your Company Website",
  },
  signUpButton: {
    id: "homescreen.signUpButton.label",
    defaultMessage: "Sign Up Now",
  },
  uploadButton: {
    id: "homescreen.uploadButton.label",
    defaultMessage: "Upload Now",
  },
});
