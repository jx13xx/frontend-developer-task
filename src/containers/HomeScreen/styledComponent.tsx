import styled from "styled-components";

export const UserLoginPage = styled.div`
  background-color: #242424;
  display: flex;
  flex-direction: row;
  width: 100%;
  height: 100vh;
  position: fixed;
`;

export const UserForm = styled.div`
  display: flex;
  flex-direction: column;
  margin-left: 103px;
  margin-top: 158px;
`;
export const UserFillUpForm = styled.div`
  display: flex;
  flex-direction: column;
`;

export const UserInputBlock = styled.div`
  display: flex;
  flex-direction: column;
  padding-bottom: 20px;
`;

export const CoverImage = styled.img`
  pointer-events: none;
  position: absolute;
  bottom: -15%;
  right: -10%;
  height: 800px;
  overflow: hidden;
  @media screen and (max-width: 842px) {
    display: none;
  }
`;

export const UserImageInput = styled.input`
  display: none;
  width: 100%;
`;
