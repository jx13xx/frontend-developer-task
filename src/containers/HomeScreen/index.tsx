import {
  HeaderUserPage,
  InputField,
  Label,
  RobotText,
  SubmitButton,
  Subtitle,
  Tagline,
} from "../../design-system/components";
import {
  CoverImage,
  UserFillUpForm,
  UserForm,
  UserInputBlock,
  UserLoginPage,
} from "./styledComponent";
import BannerImage from "images/Accent-Elements.svg";
import { injectIntl } from "react-intl";
import messages from "./messages";
import { any, InferProps } from "prop-types";

function HomeScreen({ intl }: HomeScreenTypes) {
  return (
    <UserLoginPage>
      <UserForm id="user-form">
        <HeaderUserPage id="user-form-header">
          <RobotText>{intl.formatMessage(messages.headingText)}</RobotText>
          <Subtitle>{intl.formatMessage(messages.descriptionText)}</Subtitle>
        </HeaderUserPage>

        <UserFillUpForm>
          <UserInputBlock>
            <Label>{intl.formatMessage(messages.fullNameLabel)}</Label>
            <InputField
              type="text"
              placeholder={intl.formatMessage(messages.fullNamePlaceHolder)}
            />
          </UserInputBlock>

          <UserInputBlock>
            <Label>{intl.formatMessage(messages.emailAddressLabel)}</Label>
            <InputField
              type="text"
              placeholder={intl.formatMessage(messages.emailAddressPlaceHolder)}
            />
          </UserInputBlock>
          <UserInputBlock>
            <Label>{intl.formatMessage(messages.companyLabel)}</Label>
            <InputField
              type="text"
              placeholder={intl.formatMessage(messages.companyLabelPlaceHolder)}
            />
          </UserInputBlock>
          <UserInputBlock>
            <Label>{intl.formatMessage(messages.companyWebsiteLabel)}</Label>
            <InputField
              type="text"
              placeholder={intl.formatMessage(
                messages.companyWebsitePlaceHolder
              )}
            />
          </UserInputBlock>

          <SubmitButton>
            {intl.formatMessage(messages.signUpButton)}
          </SubmitButton>
        </UserFillUpForm>
      </UserForm>
      <CoverImage src={BannerImage} />
    </UserLoginPage>
  );
}

const HomeScreenPropTypes = {
  intl: any,
};

type HomeScreenTypes = InferProps<typeof HomeScreenPropTypes>;

HomeScreen.propTypes = HomeScreenPropTypes;

export default injectIntl(HomeScreen);
