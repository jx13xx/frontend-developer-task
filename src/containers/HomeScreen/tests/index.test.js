import React from "react";
import { render } from "@testing-library/react";
import HomeScreen from "../index";
import { IntlProvider } from "react-intl";

describe("<HomeScreen/>", () => {
  it("should render and match the snapshot", () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <HomeScreen />
      </IntlProvider>
    );

    expect(firstChild).toMatchSnapshot();
  });
});
