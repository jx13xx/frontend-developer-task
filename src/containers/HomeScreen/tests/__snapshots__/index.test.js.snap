// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen/> should render and match the snapshot 1`] = `
<div
  class="sc-iqcoie ebqsGN"
>
  <div
    class="sc-crXcEl kQasWi"
    id="user-form"
  >
    <div
      class="sc-dkzDqf hvbjDy"
      id="user-form-header"
    >
      <text
        class="sc-jSMfEi gDoUFc"
      >
        Become a Member
      </text>
      <text
        class="sc-ftvSup dtRnek"
      >
        Enter your valid info to get registered
      </text>
    </div>
    <div
      class="sc-evZas gHCSVc"
    >
      <div
        class="sc-breuTD ggFkQp"
      >
        <label
          class="sc-eCYdqJ dJSjNS"
        >
          Full Name
        </label>
        <input
          class="sc-hKMtZM lpsjZE"
          placeholder="Your full name here"
          type="text"
        />
      </div>
      <div
        class="sc-breuTD ggFkQp"
      >
        <label
          class="sc-eCYdqJ dJSjNS"
        >
          Email Address
        </label>
        <input
          class="sc-hKMtZM lpsjZE"
          placeholder="Your email address here"
          type="text"
        />
      </div>
      <div
        class="sc-breuTD ggFkQp"
      >
        <label
          class="sc-eCYdqJ dJSjNS"
        >
          Company
        </label>
        <input
          class="sc-hKMtZM lpsjZE"
          placeholder="Your Company"
          type="text"
        />
      </div>
      <div
        class="sc-breuTD ggFkQp"
      >
        <label
          class="sc-eCYdqJ dJSjNS"
        >
          Company Website
        </label>
        <input
          class="sc-hKMtZM lpsjZE"
          placeholder="Your Company Website"
          type="text"
        />
      </div>
      <button
        class="sc-gsnTZi hAmfdA"
      >
        Sign Up Now
      </button>
    </div>
  </div>
  <img
    class="sc-ksZaOG dNCRCm"
    src="[object Object]"
  />
</div>
`;
